Reference/Routing/Request-Pipeline/IContentFinder.md

```diff
@@ -91,6 +91,33 @@ this example shows how to add custom content finder to (and how to remove Conten
 
 #NotFoundHandlers
 
-To set your own 404 finder create an IContentFinder and set it as the ContentLastChanceFinder.  A ContentLastChanceFinder will always return a 404 status code. Example:
+To set your own 404 finder create an IContentFinder and set it as the ContentLastChanceFinder. A ContentLastChanceFinder will always return a 404 status code. This example creates a new implementation of the IContentFinder and checks whether the requested content could not be found by using the default `Is404` property presented in the `PublishedContentRequest` class.
+
+    public class My404ContentFinder : IContentFinder {
+    	public bool TryFindContent(PublishedContentRequest contentRequest) {
+            //logic to find your 404 page and set it to contentRequest.PublishedContent
+	     CultureInfo culture = null;
+            if (contentRequest.HasDomain) {
+                culture = CultureInfo.GetCultureInfo(contentRequest.UmbracoDomain.LanguageIsoCode);
+            }
+
+            //replace 'home_doctype_alias' with the alias of your homepage
+            IPublishedContent rootNode = UmbracoContext.Current.ContentCache.GetByXPath("//home_doctype_alias").FirstOrDefault(n => n.GetCulture().ThreeLetterWindowsLanguageName == culture.ThreeLetterWindowsLanguageName);
+            //replace '404_doctype_alias' with the alias of your 404 page
+            IPublishedContent notFoundNode = UmbracoContext.Current.ContentCache.GetByXPath("//404_doctype_alias").FirstOrDefault(n => n.GetCulture().ThreeLetterWindowsLanguageName == culture.ThreeLetterWindowsLanguageName);
```

Comment:
Same remark here : using `//` is just a full content scan which is not really best practice.  This is working code, so we should not let people suggest it.  
In this case I would rather use something like `GetByXpath("root/homeDocType[id=123]/404_doctype_alias") where 123 is the Id of the root node found earlier.  Or just get all the 404 pages and then filter by the correct culture.